include/camp/resource/cuda.hpp

```diff
@@ -85,15 +85,26 @@ namespace resources
         return streams[num % 16];
       }
 
+    private:
+      Cuda(cudaStream_t s, int dev=0) : stream(s), device(dev) {}
     public:
-      Cuda(int group = -1, int device = 0) : stream(get_a_stream(group)) {}
+      Cuda(int group = -1, int dev=0) : stream(get_a_stream(group)), device(dev) {}
 
       // Methods
       Platform get_platform() { return Platform::cuda; }
       static Cuda &get_default()
       {
-        static Cuda h;
-        return h;
+        static cudaStream_t s;
+        static std::once_flag m_onceFlag;
+        std::call_once(m_onceFlag, [] {
+#if CAMP_USE_PLATFORM_DEFAULT_STREAM
+          s = 0;
+#else
+          cudaStreamCreate(&s);
+#endif
+        });
```

Comment:
I think the explicit once_flag can be avoided here.
```suggestion
        static cudaStream_t s = [] {
          cudaStream_t s_ = 0;
#if CAMP_USE_PLATFORM_DEFAULT_STREAM
          cudaStreamCreate(&s_);
#endif
          return s_;
        }();
```